### PR TITLE
Cassandra performance: repair_id prepared statements against repair_id & cache sequence

### PR DESCRIPTION
**Pull request by michaelsembwever**
Use prepared statements against repair_id, and cache a floor for the sequence number to reduce lookups.

ref: https://github.com/thelastpickle/cassandra-reaper/issues/94

### CODE REVIEW

File: src/main/java/com/spotify/reaper/storage/CassandraStorage.java [hunk 1]
```diff
@@ -707,26 +720,33 @@ public boolean updateRepairSchedule(RepairSchedule newRepairSchedule) {
...
+      ResultSet casResult = session.execute(updateRepairId.bind(idValue, idType, (idValue-1)));
       if(casResult.wasApplied()){
-        break;
+          break;
+      }else{
+          idValue = session.execute(selectRepairId.bind(idType)).one().getLong("id");
+          Preconditions.checkState(idValue < Long.MAX_VALUE);
+          attempts++;
+          if(10 <= attempts && 0 == attempts % 10){
+              LOG.warn("still cant find a new repairId after " + attempts + " attempts");
```
Comment (adejanovski) [inline src/main/java/com/spotify/reaper/storage/CassandraStorage.java:742]: MAJOR Use the built-in formatting to construct this argument. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2629)

**Comment by adejanovski** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by michaelsembwever** [comment]
Tested against creating (but not activating) a non-incremental repair with 1467 segments.

Before there was 14179 cql requests, after the patch only 5915 cql requests.

File: src/main/java/com/spotify/reaper/storage/CassandraStorage.java (hunk 1 above)
Comment (michaelsembwever) [reply src/main/java/com/spotify/reaper/storage/CassandraStorage.java:742]: of little concern. WARN is generally always enabled, and the frequency of this log line low.

Comment (tlpsonarqube) [inline src/main/java/com/spotify/reaper/storage/CassandraStorage.java:742]: MAJOR Use the built-in formatting to construct this argument. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2629)

**Comment by tlpsonarqube** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by adejanovski** [comment]
LGTM, merging.

**Pull request merged**